### PR TITLE
Uses cmake instead of build2

### PR DESCRIPTION
**Pull request by zcorniere**
The C++20 modules are currently not enough supported by the compiler to use them today. And since the main advantage in using build2 was the modules, I think that coming back to a more standard build system like `cmake` is justified.

Moving to `cmake` allow for easier integration of dependencies, as most of the libraries are build using `cmake`.

But we should keep an eye open for evolution in the support of the C++ modules and build2 update, as we may switch back to it later in the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,37 +1,10 @@
...
-b
-```
-
-## Other
-
-### Create `compile_commands.json`
-
-``` sh
-b -vn clean update |& compiledb
+## Build
```
Comment (Ersikan) [inline README.md:7]: Specify how to compile concurrently

File: libgraphics/shaders/CMakeRule.txt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+######################
+# SHADER COMPILATION #
+######################
+function(add_shader TARGET SHADER)
```
Comment (Ersikan) [inline libgraphics/shaders/CMakeRule.txt:4]: Can you add an example showing how to use this function ?

File: .clang-format [hunk 3]
```diff
@@ -0,0 +1,55 @@
+---
```
Comment (Ersikan) [inline .clang-format:1]: Why is there three dashes at the top ?

**Review by Ersikan — CHANGES_REQUESTED** [review]

File: .clang-format (hunk 3 above)
Comment (zcorniere) [reply .clang-format:1]: It YAML standard for new file

File: libgraphics/shaders/CMakeRule.txt (hunk 2 above)
Comment (zcorniere) [reply libgraphics/shaders/CMakeRule.txt:4]:
```suggestion
# add_shader(${TARGET} shader.(vert|frag|comp))
function(add_shader TARGET SHADER)
```

File: README.md (hunk 1 above)
Comment (zcorniere) [reply README.md:7]:
```suggestion
## Build
```sh 
mkdir build && cd build && cmake .. && cmake --build .  --parallel 4
```

**Pull request merged**